WordPress/src/main/java/org/wordpress/android/util/EncryptionUtils.java

```diff
@@ -0,0 +1,121 @@
+        byte[] publicKeyBytes = Base64.decode(PUBLIC_KEY_BASE64, Base64.DEFAULT);
+        NaCl.sodium().crypto_box_seal(encryptedKey, key, XCHACHA20POLY1305_KEYBYTES, publicKeyBytes);
+
+        encryptedLogJson.put("encryptedKey", Base64.encodeToString(encryptedKey, BASE64_ENCODE_FLAGS));
+
+        // Set up a new stream
+        byte[] state = new byte[XCHACHA20POLY1305_STATEBYTES];
+        byte[] header = new byte[XCHACHA20POLY1305_HEADERBYTES];
+        NaCl.sodium().crypto_secretstream_xchacha20poly1305_init_push(state, header, key);
+
+        encryptedLogJson.put("header", Base64.encodeToString(header, BASE64_ENCODE_FLAGS));
+
+        // Encrypt each log line individually and add them to JSON array
+        String logText = AppLog.toPlainText(context);
+        String[] logLines = logText.split("\n");
+        JSONArray encryptedLogLinesJson = new JSONArray();
+        int[] clen = new int[0];
+        byte[] ad = new byte[0];
+
+        for (String logLine : logLines) {
+            logLine = logLine + "\n"; // Add linebreak back
+            byte[] logLineBytes = logLine.getBytes();
+            byte[] encryptedLogLine = new byte[logLineBytes.length + XCHACHA20POLY1305_ABYTES];
+
+            NaCl.sodium().crypto_secretstream_xchacha20poly1305_push(
+                state,
+                encryptedLogLine,
+                clen,
+                logLineBytes,
+                logLineBytes.length,
+                ad,
+                0,
+                (short) 0);
+
+            encryptedLogLinesJson.put(Base64.encodeToString(encryptedLogLine, BASE64_ENCODE_FLAGS));
+        }
+
+        // Last element in the JSON array is an encrypted and encoded empty string with the FINAL tag
+        String emptyString = new String();
+        byte[] encryptedLogLine = new byte[XCHACHA20POLY1305_ABYTES];
+
+        NaCl.sodium().crypto_secretstream_xchacha20poly1305_push(
+            state,
+            encryptedLogLine,
+            clen,
+            emptyString.getBytes(),
+            0,
+            ad,
+            0,
+            (short) NaCl.sodium().crypto_secretstream_xchacha20poly1305_tag_final());
+
+        encryptedLogLinesJson.put(Base64.encodeToString(encryptedLogLine, BASE64_ENCODE_FLAGS));
+
+        encryptedLogJson.put("messages", encryptedLogLinesJson);
+
+        // Write output files for testing
+        try {
+            String path = Environment.getExternalStorageDirectory().getAbsolutePath() + "/";
+
+            BufferedWriter logOutput = new BufferedWriter(new FileWriter(path + "app_log.txt"));
+            logOutput.write(logText);
+            logOutput.close();
+
+            BufferedWriter encryptedLogOut = new BufferedWriter(new FileWriter(path + "app_log_encrypted.json"));
+            encryptedLogOut.write(encryptedLogJson.toString(4));
+            encryptedLogOut.close();
+
+            ToastUtils.showToast(context, "EncryptionUtils test code, test files saved to: " + path);
```

Comment:
While I understand that this is still testing code, we can't merge it into the repo as-is, because utility classes shouldn't be able to present UI. I think using unit tests to test this code will resolve this issue :)

Reply:
Understood.  I will remove this once I create unit tests for this functionality.

Reply:
This has been removed here: 77d4e2eb68ad1c3e87a6c1531c27e41f5d70fe90

Reply:
this has been removed here: 77d4e2eb68ad1c3e87a6c1531c27e41f5d70fe90